### PR TITLE
Implementation of DIB-R in kaolin core

### PR DESCRIPTION
**Pull request by TommyX12**
## Implementation of DIB-R in kaolin core
- [x] Copy dib-r implementation from `example` to `kaolin.graphics`
- [x] Get initial example working
  (Tested on Ubuntu 16.04, CUDA 10.0, PyTorch 1.3.1)
- [x] Clean-up and code improvement
  - [x] Remove old DIB-R code in example

### CODE REVIEW

File: kaolin/graphics/dib_renderer/rasterizer/rasterizer.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+from __future__ import print_function
+from __future__ import division
+
+import torch
+import torch.nn
+import torch.autograd
+from torch.autograd import Function
+
+import kaolin.graphics.dib_renderer.cuda.rasterizer as cuda_rasterizer
```
Comment (Caenorst) [inline kaolin/graphics/dib_renderer/rasterizer/rasterizer.py]: This line makes my `import kaolin` fails, please use:
`from ..cuda import rasterizer as cuda_rasterizer`
or (but I would prefer relative import)
`from kaolin.graphics.dib_renderer.cuda import rasterizer as cuda_rasterizer`

File: examples/renderers/DIB-R/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+results/
```
Comment (Caenorst) [inline examples/renderers/DIB-R/.gitignore]: Not a big fan of multiple .gitignore can you put this in the root .gitignore ?

File: kaolin/graphics/dib_renderer/utils/mesh.py [hunk 3]
```diff
@@ -365,51 +454,50 @@ def saveobjscale(meshfile, scale, maxratio, shift=None):
...
+
     meshfp.close()
     meshfp2.close()
-    
+
     return
 
 
-################################################################3    
+# 3
```
Comment (Caenorst) [inline kaolin/graphics/dib_renderer/utils/mesh.py]: What is this commented out 3 ?

**Review by Caenorst — CHANGES_REQUESTED** [review]
Thanks for the work, please fix the issues in the review and we are good to go :)

File: kaolin/graphics/dib_renderer/utils/mesh.py (hunk 3 above)
Comment (TommyX12) [reply kaolin/graphics/dib_renderer/utils/mesh.py]: My guess was that the original code had 3 as a typo ("#" is shift+3), and the auto formatter removed all extra "#"s. Nice catch :D removed.

File: kaolin/graphics/dib_renderer/rasterizer/rasterizer.py (hunk 1 above)
Comment (TommyX12) [reply kaolin/graphics/dib_renderer/rasterizer/rasterizer.py]: Done; Also fixed other relative imports

File: examples/renderers/DIB-R/.gitignore (hunk 2 above)
Comment (TommyX12) [reply examples/renderers/DIB-R/.gitignore]: Done.

**Review by Caenorst — APPROVED** [review]
LGTM ! Thanks for the PR !

**Pull request merged**